sklearn/cross_validation.py

```diff
@@ -259,6 +259,10 @@ def __init__(self, n, n_folds, indices, shuffle, random_state):
         if not isinstance(shuffle, bool):
             raise TypeError("shuffle must be True or False;"
                             " got {0}".format(shuffle))
+
+        if random_state is not None and not shuffle:
+            warnings.warn('shuffle must be True if you need a non default random_state.')
```

Comment:
AFAIK @amueller _BaseKFold is inherited by KFold and StratifiedKFold and both call _BaseKFold __init__ function. Should I move the warning to these classes instead?
Do you want me to add to those classes that @ragv lists?